### PR TITLE
added compact Docker Image for build

### PR DESCRIPTION
**Pull request by madduci**
:warning: This is a Pull Request Template :warning:

> Checkout all the things you've done by placing an `x` within the square braces below.

## Proposed Changes
- [X] I've read the [CODE OF CONDUCT](https://github.com/my-first-pr/hacktoberfest-2018/blob/master/CODE_OF_CONDUCT.md) and abide to it.
- [X] I've read the [CONTRIBUTING.md](https://github.com/my-first-pr/hacktoberfest-2018/blob/master/CONTRIBUTING.md)
- [X] I've forked the repository.
- [X] I've created a branch and made my changes in it.
- [X] I understand opening a PULL REQUEST doesn't mean it will be merged for sure.

It includes latest CMake and Conan.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -77,6 +77,10 @@ ARG APT="apt-get -qq --no-install-recommends"
 
 # docker build --build-arg NO_SSL="False" APT="apt-get -qq --no-install-recommends" .
 ARG NO_SSL="True"
+ARG APT="apt-get -qq --no-install-recommends"
```
Comment (blockspacer) [inline Dockerfile]: why ARG APT repeated twice?

**Review by blockspacer — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (madduci) [reply Dockerfile]: Oh i didn't see that

**Review by blockspacer — APPROVED** [review]

**Comment by madduci** [comment]
I've added a compact Docker image that can be build as 

`docker build -t devenv:llvm6 -f Dockerfile.compact .`

Featuring latest CMake and Conan, removing for the moment the need of all packages installed and then removed in the main Dockerfile. When a package is added in one layer, the file removed in another layer *do not cleanup space**.

The goal of this compact image is to be used as reference environment for a CI/CD purpose (e.g. Travis CI can run docker images)

**Comment by blockspacer** [comment]
Thanks!

**Pull request merged**